agave/fastapi/rest_api.py

```
@@ -0,0 +1,417 @@
+                path,
+                summary=f'{cls.__name__} - Query',
+                response_model=QueryResponse,
+                description=query_description,
+                responses=json_openapi(200, 'Successful Response', examples),
+                include_in_schema=include_in_schema,
+            )
+            @copy_attributes(cls)
+            async def query(
+                query_params: cls.query_validator = Depends(  # type: ignore
+                    validate_params
+                ),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)